File: DataStructures/trees/traversal/tvst.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

# In-order traversal to print values
def in_order_traversal(node):
    if node is None:
        return
    in_order_traversal(node.left)
    print(node.data, end=" ")
    in_order_traversal(node.right)

# Binary tree setup
root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)

in_order_traversal(root)
print()

File: DataStructures/trees/traversal/tvst.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* left;
    struct Node* right;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->left = NULL;
    node->right = NULL;
    return node;
}

// In-order traversal to print values
void inOrderTraversal(struct Node* node) {
    if (node == NULL) return;
    inOrderTraversal(node->left);
    printf("%d ", node->data);
    inOrderTraversal(node->right);
}

int main() {
    struct Node* root = newNode(1);
    root->left = newNode(2);
    root->right = newNode(3);
    root->left->left = newNode(4);
    root->left->right = newNode(5);

    inOrderTraversal(root);
    printf("\n");
    
    return 0;
}
